protocols/virtualfund/virtualfund.go

```diff
@@ -38,8 +39,9 @@ type GuaranteeInfo struct {
 	GuaranteeDestination types.Destination
 }
 type Connection struct {
-	Channel       *channel.TwoPartyLedger // todo: #420 deprecate in favor of consensus_channel.ConsensusChannel
-	GuaranteeInfo GuaranteeInfo
+	Channel          *channel.TwoPartyLedger // todo: #420 deprecate in favor of
+	ConsensusChannel *consensus_channel.ConsensusChannel
```

Comment:
Does it make sense to hold a `ConsensusChannel` here? Would holding something that implement this interface make more sense?
```
type LedgerChannel interface {
  IncludeGuarantee(p Proposal, sk []byte) (SideEffects, error)
  ReceiveProposal(p SignedProposal) (SideEffects, error)
  Funds(g Guarantee) bool
  Proposed(g Guarantee) bool
  // alternatively
  Status(g Guarantee) InclusionStatus
}
```

My understanding is that a `ConsensusChannel` value would not provide the functionality that `virtualfund` would need.

On the other hand, both a `LeaderChannel` and a `FollowerChannel` can implement the `LedgerChannel` interface. In this case, the `virtualfund` objective can:
- receive proposals in `Update` by calling `ReceiveProposal(p)` when a message contains `p`
- in `Crank`, check the state of ledger funding using `Funds`/`Proposed`/`Status`, and use `IncludeGuarantee` when appropriate.

Reply:
Possibly!

My intention had been to make `Connection` a working boundary between `virtualfund` and `consensus_channel`, give it (some approximation of) the above API methods, and have it internally manage delegation to the concrete types.

Reply:
Ok! I think I understand your approach, which could also work (I don't understand yet how that would work, but that's fine!)

Reply:
Colin, I would be really interested to see your approach in more detail. I played around with the `LedgerChannel` interface, and that approach does seem to encapsulate the leader/follower logic nicely outside of the virtual fund protocol logic.